### ISSUE TITLE
lint: stylish reporter to match ESLint stylish formatter

### ISSUE THREAD

**Issue by shellscape** [oxc-project/oxc#8422]
I've been experimenting with moving a rather large monorepo over to oxlint and I must say, very impressive thus far. I'd like to request a reporter to match the [ESLint stylish reporter](https://eslint.org/docs/latest/use/formatters/#stylish).

**Comment by Boshen** [comment]
Reporters can now be added to https://github.com/oxc-project/oxc/tree/main/apps/oxlint/src/output_formatter

**Comment by shellscape** [comment]
@Boshen with the help of Claude I've got a compiling reporter that passes tests. but there doesn't seem to be a good way to run formatters for development so the output can be viewed in the terminal. At least, none of the contributing docs cover how to do that. As someone new to Rust, it's not obvious what path I should take to verifying this reporter is giving me the output I want visually before submitting a PR (which is irresponsible and I would not do)

**Comment by Sysix** [comment]
hello @shellscape we updated the reporters which include some simple tests for the beginning.
You can look how we implemented each one with tests here: https://github.com/oxc-project/oxc/tree/main/apps/oxlint/src/output_formatter

There are plans to create snapshots for different CLI options, but this needs a bit more refactoring.

**Comment by shellscape** [comment]
@Sysix thanks, but not quite what I was asking. @Boshen replied to me on Discord with this:

```
https://github.com/oxc-project/oxc/blob/4ff6e8513749dfc2cc6b383b1798076580916f2a/justfile#L67-L69
just watch-oxlint test.js
```

That's pretty much what I was looking for, let's me visually inspect the reporter output. That would also make a great addition to your docs.

**Comment by drvn-mr** [comment]
It seems the doc comment hasn't been updated for the CLI help message.
https://github.com/oxc-project/oxc/blob/main/apps/oxlint/src/command/lint.rs#L188

I also notice that the line numbers are way off.

**Comment by shellscape** [comment]
@drvn-mr I was testing with `just watch` against real files and the line numbers matched with errors and warnings intentionally placed. would help to get a reproduction with input/output/expectations.

**Comment by shellscape** [comment]
Thanks. I'll see about a new PR today to address that and the help message
